### ISSUE TITLE
Port numbers

### ISSUE THREAD

**Issue by anikfal** [Unidata/awips2#698]
The summary of the output of `netstat -tpln` is as follows:
```
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name    
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      1537/sshd           
tcp        0      0 0.0.0.0:5432            0.0.0.0:*               LISTEN      12154/postmaster             
tcp        0      0 0.0.0.0:9581            0.0.0.0:*               LISTEN      13305/java              
tcp6       0      0 :::8180                 :::*                    LISTEN      12958/java                    
tcp6       0      0 :::22                   :::*                    LISTEN      1537/sshd           
tcp6       0      0 :::5432                 :::*                    LISTEN      12154/postmaster    
tcp6       0      0 127.0.0.1:15003         :::*                    LISTEN      12401/java          
tcp6       0      0 127.0.0.1:15004         :::*                    LISTEN      12825/java          
tcp6       0      0 127.0.0.1:15005         :::*                    LISTEN      12861/java          
tcp6       0      0 127.0.0.1:15006         :::*                    LISTEN      12830/java          
tcp6       0      0 :::5672                 :::*                    LISTEN      12958/java          
```
How could I know which port number is associated with which AWIPS service? For example, which service is listening via the port 5672?

Furthermore, the VM I'm using has a limited range of available ports which can be assigned to services. Many ports listed above (like 9582) are not open by the firewall. May I know if there is any configuration file by which I can set new port numbers instead of the default port numbers of the AWIPS services?

**Comment by srcarter3** [comment]
First, I might ask why you are running your own EDEX?

Running EDEX within a container (such as a virtual machine) can pose many problems and we don't recommend doing that.

Is there some reason you don't use our publicly available EDEX servers?

**Comment by anikfal** [comment]
The VM I'm using is not on a local machine, but is provided by a Cloud Service with high scalability and performance.

We have a large data archive which has been mounted on the VM, so it can be the data feed for EDEX.

I'm planning to set up the AWIPS software on this VM to provide the visualization and analysis tools for multiple users.

**Comment by tiffanycmeyer13** [comment]
First I recommend separating CAVE from EDEX if you have them on the same virtual machine. Second if you are running CAVE from a VM in the cloud you will need some type of "native" display replicator like VNC or NO MACHINE to run a desktop for each VM.

Also I'd like to mention that if you are having multiple users connect CAVE to EDEX at the same time, each user will need their own install/instance of CAVE.

**Comment by anikfal** [comment]
So AWIPS is not much similar to THREDDS? My purpose is to make CAVE available by a web-based interface, so each user can operate CAVE by their own browser.

**Comment by srcarter3** [comment]
AWIPS has multiple applications encapsulated within it, CAVE, EDEX, and python-awips.
Please refer to this video for a brief breakdown of the differences:
https://youtu.be/4ax6SRyaYTc?si=goawOUPZYQIOhK17

The THREDDS Data Server (TDS) is a web server that provides metadata and data access for scientific datasets, using a variety of remote data access protocols. You can learn more by checking out [our documentation here](https://www.unidata.ucar.edu/software/tds/).

How exactly were you planning on making CAVE available through a web browser?

**Comment by anikfal** [comment]
Regarding the THREDDS Data Server, actually I have already set it up and it's connected to our local data storage.

I saw AWIPS as a service similar to [Giovanni](https://giovanni.gsfc.nasa.gov/giovanni/), with visualization and analysis tools.

I try to focus on EDEX. It seems most CAVE tools can be accessible through Python-AWIPS by the SSH connection. So in this case no need to install CAVE for any user.
